### PR TITLE
Adding feature for bucket per service

### PR DESCRIPTION
**Pull request by SiniWilson**
### Requirements

This PR will helps to include the feature bucket per service

### CODE REVIEW

File: jenkins-build-pack-website/Jenkinsfile [hunk 1]
```diff
@@ -47,212 +46,210 @@ node()  {
...
 				events.sendFailureEvent('VALIDATE_PRE_BUILD_CONF', ex.getMessage(), context_map)
 				error ex.getMessage()
 			}
 			events.sendCompletedEvent('VALIDATE_PRE_BUILD_CONF', 'pre-build validation completed', context_map)
 		}
 
-		def s3StgBucket  = configLoader.JAZZ.S3.WEBSITE_STG_BUCKET;
-		def s3ProdBucket = configLoader.JAZZ.S3.WEBSITE_PROD_BUCKET;
-		def s3DevBucket  = configLoader.JAZZ.S3.WEBSITE_DEV_BUCKET;
+		if (configLoader.JAZZ.BUCKET_PER_SERVICE == "true") {
```
Comment (suryajak) [inline jenkins-build-pack-website/Jenkinsfile]: Can you move lines 126-143 into its own function?

File: jenkins-build-pack-website/Jenkinsfile [hunk 2]
```diff
@@ -47,212 +46,210 @@ node()  {
...
+				sh "aws configure set profile.cloud-api.region ${configLoader.AWS.REGION}"
+				sh "aws configure set profile.cloud-api.aws_access_key_id $AWS_ACCESS_KEY_ID"
+				sh "aws configure set profile.cloud-api.aws_secret_access_key $AWS_SECRET_ACCESS_KEY"
+				sh "aws configure set preview.cloudfront true"
+
+				def assetInfo = "S3 Folder - /${asset_info}"
+
+				echo "Checking if the website already exists.."
+				
+				if (configLoader.JAZZ.BUCKET_PER_SERVICE == "true") {
```
Comment (suryajak) [inline jenkins-build-pack-website/Jenkinsfile]: There is a lot of overlapping code between If and else? Can you please take a look at both and refactor it in such a way that we can drive the code through a common function with the right arguments?

File: jenkins-build-pack-website/Jenkinsfile [hunk 3]
```diff
@@ -47,346 +46,335 @@ node()  {
...
+						echo "CLOUDFRONT UPDATED"
+						echo "You can access the website at $url"					
+					} else {
+						def distributionID = getDistributionId()
+						if (distributionID != null && distributionID != "") {
+							cleanupCloudFrontDistribution()
 						}
+						generateBucketPolicy(false)
+						sh "aws s3 website s3://$s3Bucket --index-document index.html --error-document error.html"
+						url = "http://${s3Bucket}.s3-website-us-west-2.amazonaws.com/${environment}/index.html"
```
Comment (suryajak) [inline jenkins-build-pack-website/Jenkinsfile]: us-west-2 => region from config

File: jenkins-build-pack-website/Jenkinsfile [hunk 4]
```diff
@@ -47,346 +46,335 @@ node()  {
...
+						echo "You can access the website at $url"
+
+						def old_url = "https://s3.amazonaws.com/${asset_info}/index.html"   // JUST for testing
```
Comment (suryajak) [inline jenkins-build-pack-website/Jenkinsfile]: what is old_url for?

Comment (SiniWilson) [reply jenkins-build-pack-website/Jenkinsfile]: I will remove it. just for testing.

File: jenkins-build-pack-website/Jenkinsfile (hunk 2 above)
Comment (SiniWilson) [reply jenkins-build-pack-website/Jenkinsfile]: done

**Comment by rajeevr2715** [comment]
Looks fine.

**Pull request merged**